.github/workflows/publish.yaml

```diff
@@ -0,0 +1,103 @@
+name: publish
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - 
+        name: Get tags
+        id: get_tag
+        run: echo ::set-output name=TAG::${GITHUB_REF#refs/tags/}
+      -
+        name: Get Owner name
+        id: get_owner_name
+        run: echo ::set-output name=OWNER_NAME::$(echo "$GITHUB_REPOSITORY" | awk -F / '{print tolower($1)}' | sed -e "s/:refs//")
+      -
+        name: Get Repository name
+        id: get_repo_name
+        run: echo ::set-output name=REPOSITORY_NAME::$(echo "$GITHUB_REPOSITORY" | awk -F / '{print tolower($2)}' | sed -e "s/:refs//")
+      - 
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - 
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - 
+        name: Login to Github Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+      -
+        name: Build and Push container images (non-root)
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm/v7,linux/arm64
+          target: release
+          build-args: |
+            VERSION=${{ steps.get_tag.outputs.TAG }}
+            GIT_COMMIT=${{ github.sha }}
+            NS=${{ steps.get_owner_name.outputs.OWNER_NAME }}
+            REPO=${{ steps.get_repo_name.outputs.REPOSITORY_NAME }}
+          push: true
+          tags: |
+            ghcr.io/${{ steps.get_owner_name.outputs.OWNER_NAME }}/${{ steps.get_repo_name.outputs.REPOSITORY_NAME }}:${{ github.sha }}
```

Comment:
I'm seeing a lot of duplication. Seems like this could be put into a similar variable and used again?

Reply:
Do you mean something like an ENV VAR?

Reply:
The same way you split the string in a step, construct the GHCR main image prefix once instead of 6+ times.